galasa-inttests-parent/dev.galasa.inttests/src/main/java/dev/galasa/inttests/compilation/AbstractCompilationLocal.java

```diff
@@ -0,0 +1,411 @@
+        }
+    }
+    
+    /*
+     * Unpacks specified archive to a temporary location, then passes back the path to the temporary location.
+     * 
+     * @param   archiveFile             path to the archive to be downloaded
+     * @return  unpackedDir             path to the directory containing the unpacked archive
+     * @see     unpackSourceToTarget
+     */
+    private Path unpackArchiveToTemp(Path archiveFile) throws IOException {
+        logger.trace("Unzipping: " + archiveFile.toString());
+        Path unpackedDir = Files.createTempDirectory("galasa.test.compilation.unpacked");
+        unpackedDir.toFile().deleteOnExit();
+        
+        unpackSourceToTarget(archiveFile, unpackedDir);
+        
+        return unpackedDir;
+        
+    }
+    
+    /*
+     * Unpacks the specified source archive to the specified target directory.
+     * 
+     * @param   source  the archive to be unpacked
+     * @param   target  the directory to unpack the source to
+     * 
+     */
+    private void unpackSourceToTarget(Path source, Path target) throws IOException {
+        logger.trace("Unpacking: " + source.toString() + " to " + target.toString());
+        ZipInputStream zipInputStream = new ZipInputStream(Files.newInputStream(source));
+        ZipEntry entry;
+        while ((entry = zipInputStream.getNextEntry()) != null) {
+            final Path entryTarget = target.resolve(entry.getName());
+            if (entry.isDirectory()) {
+                Files.createDirectory(entryTarget);
+            } else {
+                Files.copy(zipInputStream, entryTarget);
+            }
+        }
+    }
+    
+    /*
+     * Installs gradle (Download locally, upload to image, and unzip on image), 
+     * then returns the path to the gradle bin directory. 
+     * 
+     */
+    private Path installGradle() throws ResourceUnavailableException, LinuxManagerException, IOException, IpNetworkManagerException {
+        // Download Gradle
+        logger.info("Installing Gradle");
+        logger.trace("Downloading Gradle Zip from: " + gradleZipLocation);
+        Path gradleZip = downloadHttp(gradleZipLocation);
+        Path runLocation = getLinuxImage().getHome().resolve(runName);
+        Path remoteGradleZip = runLocation.resolve("gradle-" + gradleZipVersion + ".zip");
+        Path remoteGradleDir = runLocation.resolve("gradle");
+        
+        // Upload Gradle
+        logger.trace("Uploading gradle archive to remote image");
+        logger.trace("Copying: " + gradleZip.toString() + " to " + remoteGradleZip.toString());
+        Files.copy(gradleZip, remoteGradleZip);
+        
+        // Unzip Gradle
+        logger.trace("Unzipping gradle archive on remote image");
+        String unzipRC = getLinuxImage().getCommandShell().issueCommand(
+                "unzip -q " + remoteGradleZip.toString()
+                + " -d " + remoteGradleDir.toString()
+                + "; echo $?"
```

Comment:
better if the echo is RC=$?   ensure false positives

Reply:
Check is now more verbose (checks for `RC=0`)